### PR TITLE
feat(address): entity, and crud operations

### PR DESCRIPTION
**Pull request by cchampou**

### CODE REVIEW

File: src/modules/address/dto/create-address.dto.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export class CreateAddressDto {}
```
Comment (draphikas) [inline src/modules/address/dto/create-address.dto.ts:1]: It should not be empty

File: src/modules/address/entities/address.entity.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@Entity()
+export class Address {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  addressLine1: string;
+
+  @Column()
+  addressLine2: string;
```
Comment (draphikas) [inline src/modules/address/entities/address.entity.ts:12]: do not agree with naming, I prefer 'street' and 'complement'

File: src/modules/address/entities/address.entity.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  @Column({
+    type: 'float8',
+  })
+  latitude: number;
+
+  @Column({
+    type: 'float8',
+  })
+  longitude: number;
+}
```
Comment (draphikas) [inline src/modules/address/entities/address.entity.ts:26]: Where is the city relation ?

Comment (cchampou) [reply src/modules/address/entities/address.entity.ts:26]: Was before rebasing with develop, need update :)

File: src/modules/address/dto/create-address.dto.ts (hunk 1 above)
Comment (cchampou) [reply src/modules/address/dto/create-address.dto.ts:1]: Fixed, I'd love to have another review on this :wink:

File: src/modules/address/entities/address.entity.ts (hunk 2 above)
Comment (cchampou) [reply src/modules/address/entities/address.entity.ts]: Renamed !

File: src/modules/address/entities/address.entity.ts (hunk 3 above)
Comment (cchampou) [reply src/modules/address/entities/address.entity.ts:30]: Fixed !

File: src/modules/address/dto/create-address.dto.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export class CreateAddressDto {
+  @IsString()
+  street: string;
+
+  @IsString()
+  complement: string;
+
+  @IsString()
+  zipCode: string;
```
Comment (draphikas) [inline src/modules/address/dto/create-address.dto.ts:11]: Add IsNotEmpty on all as we don't want to accept  `""`

File: src/modules/address/dto/create-address.dto.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+  @IsNumber()
+  latitude: number;
+
+  @IsNumber()
+  longitude: number;
```
Comment (draphikas) [inline src/modules/address/dto/create-address.dto.ts:17]: Replace isNumber by IsLatitude and IsLongitude

File: src/modules/address/dto/create-address.dto.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+}
```
Comment (draphikas) [inline src/modules/address/dto/create-address.dto.ts:18]: We want to link address to a city, so we need to validate that, on create, there is an object with a city id valid

See Line 10 to 12: https://github.com/cchampou/katyv4-travel-website/blob/1d008e8b917a894aeec9923ab28cae287b7a54d0/src/modules/city/dto/create-city.dto.ts#L10

File: src/modules/address/entities/address.entity.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  })
+  latitude: number;
+
+  @Column({
+    type: 'float8',
+  })
+  longitude: number;
+
+  @ManyToOne(() => City, (city) => city.addresses)
+  city: Relation<City>;
```
Comment (draphikas) [inline src/modules/address/entities/address.entity.ts:29]: Make relation not nullable

**Review by draphikas — APPROVED** [review]

**Pull request merged**